### PR TITLE
Create config for mysql replicated statefulset

### PR DESCRIPTION
**Pull request by earlgreyz**
https://github.com/grtl/mysql-operator/issues/38
https://github.com/grtl/mysql-operator/issues/42

### CODE REVIEW

File: artifacts/mysql-statefulset.yaml [hunk 1]
```diff
@@ -0,0 +1,167 @@
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: {{ .Spec.Name }}
+spec:
+  selector:
+    matchLabels:
+      app: {{ .Spec.Name }}
+  serviceName: {{ .Spec.Name }}
+  replicas: 1
```
Comment (m-chrzan) [inline artifacts/mysql-statefulset.yaml]: If we're introducing master/slave configuration, replica count should be configurable now.

File: artifacts/mysql-statefulset.yaml [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+        livenessProbe:
+          exec:
+            command: ["mysqladmin", "ping"]
+          initialDelaySeconds: 30
+          periodSeconds: 10
+          timeoutSeconds: 5
+        readinessProbe:
+          exec:
+            # Check we can execute queries over TCP (skip-networking is off).
+            command: ["mysql", "-h", "127.0.0.1", "-e", "SELECT 1"]
```
Comment (m-chrzan) [inline artifacts/mysql-statefulset.yaml]: Wouldn't this need a password? Either way, the pod doesn't start on my machine, not sure if this is the reason or something else.

File: util/template.go [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+	"text/template"
+
+	"k8s.io/apimachinery/pkg/util/yaml"
+
+	crv1 "github.com/grtl/mysql-operator/pkg/apis/cr/v1"
+)
+
+// ObjectFromTemplate executes Go template with given cluster object and
+// parses the result into the object structure.
+func ObjectFromTemplate(cluster *crv1.MySQLCluster, object interface{}, templateFile string) error {
```
Comment (m-chrzan) [inline util/template.go]: This should probably be called something like ClusterObjectFromTemplate if this is specifically for templates that need a cluster. Maybe write a more general version that we'd also be able to use with the CRD and ConfigMap?

File: artifacts/mysql-statefulset.yaml (hunk 2 above)
Comment (earlgreyz) [reply artifacts/mysql-statefulset.yaml]: For now all servers run mysql without root password:
```yaml
  containers:
      - name: mysql
        image: mysql:latest
        env:
          - name: MYSQL_ALLOW_EMPTY_PASSWORD
            value: "1"
```

**Pull request merged**